packages/js/components/src/experimental-select-control/menu.tsx

```diff
@@ -33,13 +33,19 @@ export const Menu = ( {
 	const [ boundingRect, setBoundingRect ] = useState< DOMRect >();
 	const selectControlMenuRef = useRef< HTMLDivElement >( null );
 
-	useEffect( () => {
-		if ( selectControlMenuRef.current?.parentElement ) {
+	useLayoutEffect( () => {
+		if (
+			selectControlMenuRef.current?.parentElement &&
+			selectControlMenuRef.current?.parentElement.clientWidth > 0
```

Comment:
This was needed as the block editor seems to render the blocks before they are being displayed on the screen, so this would return a bounding rect with width and height being zero every time.
This condition and the additional dependency below helps with this.